quilt/data.py

```diff
@@ -41,6 +41,12 @@ def _class_repr(self):
     def __repr__(self):
         return self._class_repr()
 
+    def __setattr__(self, name, value):
+        if name.startswith('_') or isinstance(value, Node):
+            super(Node, self).__setattr__(name, value)
+        else:
+            raise ValueError("{val} is not a valid package node".format(val=value))
```

Comment:
Actually, sorry, I was wrong earlier; I think `AttributeError` is the appropriate exception here.